### PR TITLE
Add clang-format to pre-commit

### LINKED ISSUE #868: Add make target for clang-format.
As in msprime, to save devs needing to remember which files to format.

### PR DESCRIPTION
**Pull request by benjeffery**
I've found a way to skip `clang-format` if it isn't found on the system.

Fixes #868, #891, #820

@petrelharp Mergify is configured to require 1 approving review - so while @jeromekelleher is away I'll tag you for reviews ok?

### CODE REVIEW

**Comment by AdminBot-tskit** [comment]
📖 Docs for this PR can be previewed [here](https://tskit-dev.github.io/tskit-docs/899/)

File: docs/development.rst [hunk 1]
```diff
@@ -627,14 +627,8 @@ Code style
...
 C code is formatted using
 `clang-format <https://clang.llvm.org/docs/ClangFormat.html>`_
-with a custom configuration.
-To ensure that your code is correctly formatted, you can run
-
-.. code-block:: bash
-
-    clang-format -i c/tskit/* c/tests/*.c c/tests/*.h
-
-before submitting a pull request.
```
Comment (petrelharp) [inline docs/development.rst:637]: I think it's nice to leave in the DIY command? pre-commit checks a lot of things and can produce a lot of output, so it's less overwhelming to be able to easily run them separately sometimes.  Unless this is out of date somehow?

Comment (petrelharp) [reply docs/development.rst:637]: It could say "If you have clang-format installed, you can run the check manually as follows..."

**Review by petrelharp — APPROVED** [review]
LGTM, with the minor suggestion (or not)

File: docs/development.rst (hunk 1 above)
Comment (mufernando) [reply docs/development.rst:637]: I agree with Peter. But just noticed that when you run clang-format in `c/tskit/trees.c` it changes sth that it shouldnt.

Comment (benjeffery) [reply docs/development.rst:637]: >  But just noticed that when you run clang-format in c/tskit/trees.c it changes sth that it shouldnt.
What version of `clang-format`? It is not stable across releases. CI and this pre-commit use 6.0.

Comment (mufernando) [reply docs/development.rst:637]: hah, thanks! I had 5.0.1

Comment (petrelharp) [reply docs/development.rst:637]: Oh, well if it depends precisely on the version, maybe we should just say "You can run clang-format yourself, but note that the results depend on clang-format's version, and we use a specific version in the pre-commit checks."

Comment (petrelharp) [reply docs/development.rst:637]: Or even "We don't recommend you run clang-format yourself, because the results depends on clang-format's version."

Comment (benjeffery) [reply docs/development.rst:637]: I've re-added a manual running snippet and added a note about the version.

**Comment by benjeffery** [comment]
Merging manually due to github action changes blocking mergify.

**Pull request merged**